sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClient.java

```diff
@@ -119,111 +129,226 @@
-            .map(response -> Utilities.generateResponseFromModelType(response, Utilities.attestationSignersFromJwks(response.getValue())));
+            .map(response -> Utilities.generateResponseFromModelType(response, AttestationSignerImpl.attestationSignersFromJwks(response.getValue())));
+    }
+
+
+    /**
+     * Processes an OpenEnclave report , producing an artifact. The type of artifact produced is dependent upon
+     * attestation policy.
+     *
+     * @param report - OpenEnclave report to attest.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the result of an attestation operation.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<AttestationResult>> attestOpenEnclaveWithResponse(byte[] report) {
+        return this.attestOpenEnclaveWithResponse(new AttestationOptionsImpl().setEvidence(report));
     }
 
     /**
      * Processes an OpenEnclave report , producing an artifact. The type of artifact produced is dependent upon
      * attestation policy.
      *
-     * @param request Attestation request for Intel SGX enclaves.
+     * @param options Attestation options for attesting SGX enclaves.
      * @throws IllegalArgumentException thrown if parameters fail the validation.
-     * @throws CloudErrorException thrown if the request is rejected by server.
+     * @throws HttpResponseException thrown if the request is rejected by server.
      * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
      * @return the result of an attestation operation.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<AttestationResponse>> attestOpenEnclaveWithResponse(AttestOpenEnclaveRequest request) {
-        return withContext(context -> this.attestOpenEnclaveWithResponse(request, context));
+    public Mono<Response<AttestationResult>> attestOpenEnclaveWithResponse(AttestationOptions options) {
+        return withContext(context -> this.attestOpenEnclaveWithResponse(options, context));
+    }
+
+    /**
+     * Processes an OpenEnclave report , producing an artifact. The type of artifact produced is dependent upon
+     * attestation policy.
+     *
+     * @param report OpenEnclave generated SGX report.
+     * @param context Context for request.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the result of an attestation operation.
+     */
+    public Mono<Response<AttestationResult>> attestOpenEnclaveWithResponse(byte[] report, Context context) {
```

Comment:
Same as above and the rest below.

Reply:
> > ```
> > public AttestationClientBuilder apiVersion(String apiVersion) {
> > ```
> 
> this needs to be serviceversion.
> Sample -
> 
> https://github.com/Azure/azure-sdk-for-java/blob/a804ef19f7f7d099747bd60d67d4b9bf079e2ead/sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/MetricsAdvisorClientBuilder.java#L393
> 
> Refers to: sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationClientBuilder.java:76 in [4508c5c](https://github.com/Azure/azure-sdk-for-java/commit/4508c5c8354bb6145adde3058b878400828dd8dd). [](commit_id = [4508c5c](https://github.com/Azure/azure-sdk-for-java/commit/4508c5c8354bb6145adde3058b878400828dd8dd), deletion_comment = False)

Why is it ApiVersion in the generated swagger and ServiceVersion in the hand written code? Shouldn't they be consistent?

Reply:
Also, why doesn't the CheckStyle plugin check this???

Reply:
I think APIView should show this as a potential missing field. Checkstyle can't complain about ApiVersion since it could very well be another public state of the builder, but we could add a check for serviceVersion being missing.

Reply:
It complains about a missing ServiceVersion, but that seems to be something at the package level, not the builder level (based on where the error occurs). I asked you about how to resolve the error several days ago.

Reply:
Also, changing the setter name didn't change the error in ApiView.

Reply:
Can you try changing this to serviceVersion and see if that message goes away?

Reply:
It didn't - the current ApiView has the error and the serviceVersion method.

Reply:
Huh- I will ask around about that particular one :)

Reply:
It turns out it's more than just the method name. It's also the creation of an enumeration for the method name.

This really should be in the training - it's required, but isn't mentioned. And the generated SDK gets it wrong.